Form/Type/SeoMetadataType.php

```diff
@@ -69,8 +76,15 @@ public function setDefaultOptions(OptionsResolverInterface $resolver)
      */
     public function configureOptions(OptionsResolver $resolver)
     {
+        $class = null;
+        if ($this->container->has('cmf_seo.backend_type_phpcr')) {
```

Comment:
when having a configuration option or a parameter created in the Extension class, i wouldn't need the container anymore, and yes i would use the constructor injection then.
I think creating a `seo_meta_data_class` parameter based on the persistence decission, would be the easiest way. I wouldn't make it as a configuration, why should we?